spec/element_spec.rb

```diff
@@ -75,6 +75,20 @@
 
       expect(browser.div(:id, 'text')).to_not exist
     end
+
+    it "returns appropriate value when an ancestor element becomes stale" do
+      stale_element = browser.div(id: 'top').div(id: 'middle').div(id: 'bottom')
+      expect(stale_element.present?).to be true # look up and store @element for each element in hierarchy
+
+      grandparent = stale_element.instance_variable_get('@parent').instance_variable_get('@parent').instance_variable_get('@element')
+
+      # simulate element going stale during lookup
+      allow(grandparent).to receive('enabled?') { raise Selenium::WebDriver::Error::ObsoleteElementError }
+
+      browser.refresh
+      expect(stale_element.present?).to be Watir.always_locate?
+      expect(stale_element.present?).to be true
```

Comment:
My last commit here removes all the extra reset! calls. All of the previous tests are passing. So, we can consider the previous behavior to be undocumented and unnecessary. We have the Watir.always_locate? turned on, so this won't affect us at Blackbaud, either way.